### ISSUE TITLE
Decide on a behavior if table creation fails

### ISSUE THREAD

**Issue by neumino** [neumino/thinky#289]
Should we just print something? throw an error?

**Comment by neumino** [comment]
The original behavior was I think to crash since you can't do anything if your table is not created.

We seem to catch the error at some point, but not everywhere, so the behavior is a little fuzzy at the moment.

**Comment by neumino** [comment]
We should probably ask people to add a listener for `error`.

**Comment by gjuchault** [comment]
> The original behavior was I think to crash since you can't do anything if your table is not created.

One table could fail and not the others, throwing an error is maybe the most logical way to do

**Comment by neumino** [comment]
So the problem is that you can't do much with a model that is not initialized.

We could keep retrying until we know that the model was initialized. It's not that hard, it's just some work (doable though).

**Comment by mbroadst** [comment]
@neumino technically I think you _can_ do a lot with a model that's not initialized. Thinky directly couples the table/model creation with the schema definitions, which I think can be problematic (and has been for me).

In sequelize, for instance, they default to the model definitions being just that: definitions. If you want to ensure the tables are created, you have to explicitly call "sync()" and it will modify your database, but otherwise the model definitions are simply used as metadata to inform operations in the ORM. 

I find the lack of clear separation between many of these ideas in thinky to be problematic. For example, I can't explicitly connect to the database, I have to leave it up to "magic" happening in the background with the pool, and the Model table creation logic (I only just learned today that these are two very separate and distinct code paths).

TL;DR - maybe adding support for the following would clear up some issues:
  - ability to manually call ```thinky.connect()``` to establish a connection (my rest api might call this at the beginning to ensure that we have a valid connection to rethink, and be able to make a decision if that's not possible)
  - allow users to opt-out of the requirement for automatic table generation in models (maybe providing something like sequelize has with sync())
  -  (I'm sure I'm missing something here...)

**Comment by JohnyDays** [comment]
In continuation from @mbroadst's comment, I have had necessity for testing the connection, so something like he proposed with thinky.connect() would be awesome. :+1:
